packages/react-ui/src/components/stack/index.js

```diff
@@ -15,15 +27,10 @@ const Stack = ({ inline, justify, align, direction, gap, css, ...props }) => {
     styles.flexDirection = direction.map(d =>
       d === 'vertical' ? 'column' : 'row'
     )
-    styles['> *:not(:last-child)'] = direction.map(d => ({
-      [d === 'vertical' ? 'marginBottom' : 'marginRight']: gap,
-      [d === 'vertical' ? 'marginRight' : 'marginBottom']: 0
```

Comment:
I've removed this as I don't know why this has been used here, didn't make sense but I could have missed something. If anything - I think this should either be used for none or for both of array/flat `direction`

Reply:
I want to say this is to handle the responsive syntax `[mobile, tablet, desktop]` properly, but it's been a while :)

But, I can confirm that it's needed

Reply:
Ok, I've brought this back - but I use the same logic for both branches of the code. IMHO it should be fine but please check it.